src/ShallowTraversal.js

```diff
@@ -127,9 +133,16 @@ export function buildPredicate(selector) {
       }
       break;
 
+    case 'object':
+      if (!Array.isArray(selector) && selector !== null && !isEmpty(selector)) {
+        return node => nodeMatchesObjectProps(node, selector);
+      }
+      throw new TypeError(
+        'Enzyme::Selector does not support an empty object, array, or null as a selector'
```

Comment:
Arguably, you could fence for each one of these scenarios separately and throw an error for each one. Might be more helpful to the user. Not a big deal though.